samples/bluetooth/hci_usb/README.rst

```diff
@@ -21,3 +21,28 @@ This sample can be found under :zephyr_file:`samples/bluetooth/hci_usb` in the
 Zephyr tree.
 
 See :ref:`bluetooth samples section <bluetooth-samples>` for details.
+
+Notes on HCI over USB
+*********************
+
+USB is a serial protocol. However, the implementation of the peripheral can
+expose multiple endpoints, in effect re-ordering serial data into per-endpoint
+receive queues.
+
+The Bluetooth specification defines separate endpoints per HCI packet type.
+
+As the HCI protocol was not originally designed with re-ordering in mind, using
+this transport may expose the application to deadlocks and general unexpected
+behavior.
```

Comment:
That is not quite correct. The physical layer is a serial protocol, above it is bit different. A USB function must have one or more interfaces, an interface can have zero or more endpoints. The Bluetooth HCI USB transport layer first interface consists of three endpoints: interrupt IN for HCI events, bulk IN and bulk OUT for ACL data. HCI commands are transferred through the control pipe (host-to-device only). You can think of it as three different logical channels, for HCI commands, HCI events and ACL data, using three different 'transport pipes', Control, Interrupt (periodic transfers), Bulk. How these transfer types are scheduled within a USB frame may depend on the USB controller type. There is no reordering. This is not a new problem, it was just not considered when the USB transport layer in Bluetooth was designed.

Reply:
Could you then please suggest an alternative text?

Reply:
I will make a proposal in a few days.

Reply:
I can try to come up with something myself as well

Reply:
it's been a few days 😅
jokes aside, do you have proposals for rewording my original statement? Everyone I spoke with in the Bluetooth stack team also thinks that the endpoint-based USB transport is broken.